Objects/typeobject.c

```diff
@@ -8134,225 +8127,240 @@ typedef struct wrapperbase slotdef;
 #undef BINSLOT
 #undef RBINSLOT
 
-#define TPSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
+#define TPSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
     {NAME, offsetof(PyTypeObject, SLOT), (void *)(FUNCTION), WRAPPER, \
-     PyDoc_STR(DOC)}
-#define FLSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, FLAGS) \
+     PyDoc_STR(DOC), .name_strobj = NAMEOBJ}
```

Comment:
The extra macro argument is unnecessary (and the duplication is avoidable):

```c
#define TPSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
    {#NAME, offsetof(PyTypeObject, SLOT), (void *)(FUNCTION), WRAPPER, \
     PyDoc_STR(DOC), .name_strobj = &_Py_ID(NAME)}
```

(You could also use `_Py_XSTRINGIFY()` from pymacro.h.)

Note that this requires that the quotation marks be dropped in all the uses of this macro.  I'll leave a separate comment to show an example.

Reply:
One consequence of this change is that Tools/scripts/generate_global_objects.py cannot automatically detect these names.  So the script would need to be updated.  This includes making sure all the names are in the `IDENTIFIERS` list at the top of the file.  I think this is worth doing.

Reply:
FYI, I opened gh-94715 to separately consider the idea of detecting the indirect uses of `_Py_ID()` as well.  That can be dealt with in a separate PR (if desirable), though.

Reply:
FWIW, I'd still rather we take an approach like I suggested above, instead of passing in the duplicated `_Py_ID()` as an extra macro argument.  `slotdefs[]` is already pretty bulky.  We should work to avoid reduced readability there, even if it means a little extra complexity in our tooling.

That said, the only extra cost is that we add all the strings to the list at the top of generate_global_objects.py.  That's how we're already handling these cases so I'm not suggesting something new and it's not much work.  (...and I expect we will find a satisfactory solution for that situation in gh-94715.)

Reply:
Ok done.